p/kako-nastane-spletna-stran.html

```diff
@@ -0,0 +1,403 @@
+Kaj pa na spletu?
+
+- v brskalnik **vtipkamo** ali pa **kliknemo** na URL naslov
+- brskalnik **zahtevek** opremi z dodatnimi podatki
+- zahtevek preko ogromno kanalov prileti do **strežnika**
+- ta ga obdela na podlagi **algoritma** v navodilo za brskalnik
+- strežnik vrne navodilo po kanalu do brskalnika
+- brskalnik **izriše** spletno stran uporabniku
+]
+---
+# Naročilo
+
+Kako izgleda naročilo?
+
+- "Želim bolognese pizzo z nacho sirom."
+  - "http://www.picarija.si/pizza/bolognese?dodatki=nacho"
+- "Gospa natakarica, želim vegetarjansko pizzo brez olive in nacho sirom."
+  - "http://natakarica@www.picarija.si/pizza/vege?brez=oliva&dodatki=nacho"
+- "Želim navadno pizzo, vendar varno zapakirano v škatlici, da jo odnesem domov."
+  - "https://www.picarija.si/pizza/navadna"
+- "Halo halo, Anon pri telefonu, naročil bi dunajski zrezek na naslov ..."
+  - "https://www.picarija.si:071123456/kosilo/dunajski" (najdaljša telefonska števila je sicer lahko le **65535**)
+- "Naročila bi veliko kmečko pizzo, vendar na dva krožnika."
+  - "http://www.picarija.si/pizza/kmecka?velikost=velika#kroznik=2"
+
+Resnica o URL naslovu
+```
+protokol://uporabnik:geslo@streznik:vrata/pot/do/programa?parameter1=Pozdravljen&...
+```
+---
+# Naročilo kot URL
+
+- protokol: **http**, *https*, ftp, ssh, ...
+- vrata: **80**, 443, 8080
+- uporabnik, geslo: privzeto brez
+- strežnik: IP številka ali domena
+
+## Kako od domene do IP številke
+V resnici ima vedno neko številko (IP naslov), ampak za lažje brskanje lahko pišemo imena (domene),
+ki so shranjene v nekem imeniku (DNS strežnik). Ko vtipkamo `http://www.zemanta.com` se najprej povpraša *korenski* imenik, kdo je `com`, nato kdo je `zemanta` in na koncu kdo je `www`. V odgovor dobimo neko številko, npr. `77.94.136.208`, ki je kot naslov, npr. *Celovška cesta 32*.
+
+---
+Zahtevek (naročilo bolj uradno)
+```
+GET / HTTP/1.1
+Host: course.coderdojo.si
+Connection: keep-alive
+Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
+User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)
+Chrome/33.0.1750.149 Safari/537.36
+Accept-Encoding: gzip,deflate,sdch
+Accept-Language: en-US,en;q=0.8,sl;q=0.6,en-GB;q=0.4
+Cookie: _ga=GA1.2.326423702.1393943140; __utma=238833810.326423702.1393943140.1394526334.1395071780.4;
+__utmb=238833810.1.10.1395071780; __utmc=238833810;
+__utmz=238833810.1393943140.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none)
+```
+Odgovor strežnika:
+```
+HTTP/1.1 200 OK
+Server: GitHub.com
+Content-Type: text/html; charset=utf-8
+Last-Modified: Tue, 04 Mar 2014 14:30:57 GMT
+Expires: Mon, 17 Mar 2014 16:15:49 GMT
+Cache-Control: max-age=600
+Content-Encoding: gzip
+Content-Length: 1277
+Accept-Ranges: bytes
+Date: Mon, 17 Mar 2014 16:05:49 GMT
+Via: 1.1 varnish
+Age: 0
+Connection: keep-alive
+X-Served-By: cache-fra1222-FRA
+X-Cache: MISS
+X-Cache-Hits: 0
+X-Timer: S1395072349.647996664,VS0,VE98
+Vary: Accept-Encoding
+```
+[vsebina strani ...]
```

Comment:
em, tole sem mislil znotraj code bloka. pač HTTP/1.1 request je:
```
POST / HTTP/1.1
Headers: ...

param1=val1&param2=val2&...
```
in response: 
```
HTTP/1.1 200 OK
Headers: ...

<DOCTYPE html>
<html>...
```
torej vsebina ločena od zaglavja z `\n\n`.